core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -167,7 +256,275 @@ public void setTransportSender(TransportSender transportSender) {
+                    parameter -> {
+                        if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam != null) {
+                                parameterList.add(paramValues.get(pathParam.value()));
+                            } else {
+                                throw new IllegalArgumentException("String parameters without" +
+                                                                                 " @PathParam annotation");
+                            }
+                        } else {
+                            parameterList.add(null);
+                        }
+                    }
+            );
+            executeMethod(method, routableEndpoint.getDestination(), parameterList, session);
+            return true;
+        } catch (Throwable throwable) {
+            handleError(carbonMessage, throwable, routableEndpoint, session);
+            return false;
+        }
+    }
+
+    /*
+    Handle Text WebSocket Message
+     */
+    private void handleTextWebSocketMessage(TextCarbonMessage textCarbonMessage,
+                                           PatternPathRouter.RoutableDestination<Object>
+                                                   routableEndpoint, Session session) {
+        Object endpoint = routableEndpoint.getDestination();
+        Map<String, String> paramValues = routableEndpoint.getGroupNameValues();
+        Method method = new EndpointDispatcher().getOnStringMessageMethod(endpoint);
+        try {
+            List<Object> parameterList = new LinkedList<>();
+            boolean isStringSatifsfied = false;
+            Arrays.stream(method.getParameters()).forEach(
+                    parameter -> {
+                        if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam == null) {
+                                parameterList.add(textCarbonMessage.getText());
+                            } else {
+                                if (isStringSatifsfied == false) {
+                                    parameterList.add(paramValues.get(pathParam.value()));
+                                } else {
+                                    throw new IllegalArgumentException("String parameters without" +
+                                                                                     " @PathParam annotation");
+                                }
+                            }
+                        } else if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else {
+                            parameterList.add(null);
```

Comment:
I tried it when we do it it gives us
**java.lang.IllegalArgumentException: argument type mismatch**